msg_helpers.go

```diff
@@ -314,23 +288,11 @@ func unpackString(msg []byte, off int) (string, int, error) {
 		switch b {
 		case '"', '\\':
 			s = append(s, '\\', b)
-		case '\t':
-			s = append(s, '\t')
-		case '\r':
-			s = append(s, '\r')
-		case '\n':
-			s = append(s, '\n')
+		case '\t', '\r', '\n':
+			s = append(s, b)
 		default:
 			if b < 32 || b > 127 { // unprintable
-				var buf [3]byte
-				bufs := strconv.AppendInt(buf[:0], int64(b), 10)
-				s = append(s, '\\')
-				for i := 0; i < 3-len(bufs); i++ {
-					s = append(s, '0')
-				}
-				for _, r := range bufs {
-					s = append(s, r)
-				}
+				s = append(s, fmt.Sprintf("\\%03d", b)...)
```

Comment:
Also silly benchmark could be great. On other hand

```
   if b < 9 {
       s = append (s, '\\', '0', '0', '0'+b)
   } else if b < 32 {
       s = append (s, '\\', '0', '0'+(b/10), '0'+(b%10))
   } else if b >= 200 {
       s = append (s, '\\', '2', '0'+((b/10)%10), '0'+(b%10))
   } else if b > 127 {
       s = append (s, '\\', '1', '0'+((b/10)%10), '0'+(b%10))
   }
```

Might be worth micro-benchmarking too.